### PR TITLE
Convert LWRPs to custom resources & fix the specs

### PR DESCRIPTION
**Pull request by tas50**
Convert the LWRPs to custom resources to support changes in chef 13 and beyond

### CODE REVIEW

File: resources/tap.rb [hunk 1]
```diff
@@ -19,13 +19,49 @@
...
+          String,
           regex: %r{^[\w-]+(?:\/[\w-]+)+$}
 
-attribute :tapped,
-          kind_of: [TrueClass, FalseClass]
+property :tapped,
+          [true, false]
+
+
+          include ::Homebrew::Mixin
```
Comment (Brantone) [inline resources/tap.rb]: That's some interesting leading whitespace going on.
Any particular reason?

Comment (tas50) [inline resources/tap.rb]: Just hit enter a few too many times

Comment (Brantone) [inline resources/tap.rb]: Was thinking more the horizontal spacing with lines 31 - 67 being over 10 spaces.

File: resources/tap.rb [hunk 2]
```diff
@@ -19,13 +19,47 @@
...
+      command "/usr/local/bin/brew tap #{new_resource.name}"
+      environment lazy { { 'HOME' => ::Dir.home(homebrew_owner), 'USER' => homebrew_owner } }
+      not_if "/usr/local/bin/brew tap | grep #{new_resource.name}"
+      user homebrew_owner
+    end
+  end
+end
+
+action :untap do
+  if @tap.tapped
```
Comment (rjaros87) [inline resources/tap.rb]: `@tap` ;)

Comment (tas50) [inline resources/tap.rb]: Yep that was fixed a bit later

**Pull request merged**